### ISSUE TITLE
Add option to force the dependency to be collected 

### ISSUE THREAD

**Issue by pecalleja** [RKrahl/pytest-dependency#37]
It would be nice an option in the dependency that force to collect and execute the required test. For example:
```python
import pytest

@pytest.mark.dependency()
@pytest.mark.xfail(reason="deliberate fail")
def test_a():
    assert False

@pytest.mark.dependency()
def test_b():
    pass

@pytest.mark.dependency(depends=["test_a"])
def test_c():
    pass

@pytest.mark.dependency(depends=["test_b"], collect=True)
def test_d():
    pass

@pytest.mark.dependency(depends=["test_b", "test_c"])
def test_e():
    pass
```

if execute only the `test_d` the collector also will select the `test_b` because the hard restriction of the dependency `collect=True`

If you think this is a good idea I can work on this feature. kind regards

**Comment by RKrahl** [comment]
I'm skeptical. Until now, pytest-dependency is neat and small because it has a very clear scope and that is about skipping tests. In principle, I'd prefer to keep the scope focused and keep out of neighbouring tasks such as collecting or ordering.

That does not mean that I'm totally against such an idea. But it does mean that it might take a significant effort to convince me that going that way is possible without opening a can of worms.

**Comment by pecalleja** [comment]
Yeah, I agree with you about this, let's keep this awesome project neat and small.  let me

**Comment by pecalleja** [comment]
Taking the first comment example, imagine the case when you need to set `test_d` dependency with `test_b`. One option would be to call one test into the other :
```python
import pytest

@pytest.mark.dependency()
@pytest.mark.xfail(reason="deliberate fail")
def test_a():
    assert False

@pytest.mark.dependency()
def test_b():
    pass

@pytest.mark.dependency(depends=["test_a"])
def test_c():
    pass

@pytest.mark.dependency()
def test_d():
    test_b()
    pass

@pytest.mark.dependency(depends=["test_b", "test_c"])
def test_e():
    pass
```
this solution force to execute the `test_b` but this solution let's `test_b` get out of pytest report of collected tests. also when calling the entire file, the `test_b` will be called twice...  this sorts of thing I want to resolve with an option forcing to collect the "required" test dependency.

**Comment by a-recknagel** [comment]
@RKrahl Just to add my voice, I'd also appreciate such a feature, maybe enabled with a flag that I can put in my .ini though, adding a keyword to all functions creates a lot of noise. My use case is that Pycharm gives all my tests little play-buttons that I can press to run them, which just calls the `pytest path/to/test.py::test_function` form, and is very handy. Without pulling dependencies into the collection, I'd have to restructure my tests, think, or just run the whole suite, which all incur a non-trivial cost.

@pecalleja I tried your snippet, it didn't work for me, even after changing `item = pytest.Function(name=dependency, parent=parent)` to `item = pytest.Function.from_parent(name=dependency, parent=parent)` after pytest complained about the former.

**Comment by marcinwrochna** [comment]
I had to modify the snippet as follows:
```python

@pytest.hookimpl(trylast=True)
def pytest_collection_modifyitems(config: pytest.Config, items: list[pytest.Item]) -> None:
    seen = set[str]()
    new_items = list[pytest.Item]()

    def dfs(item: pytest.Item) -> None:
        if item.name in seen:
            return
        seen.add(item.name)
        for marker in item.own_markers:
            if marker.name == "dependency" and marker.kwargs.get("collect"):
                for dependency in marker.kwargs.get("depends", []):
                    dfs(pytest.Function.from_parent(name=dependency, parent=item.parent))
        new_items.append(item)

    for item in items:
        dfs(item)

    old_names = set(item.name for item in items)
    new_names = [item.name for item in new_items if item.name not in old_names]
    if new_names:
        print("Un-deselected:", *new_names)

    items[:] = new_items
```
This assumes just simple function dependencies in the same parent module; I don't know how to do it in general, but https://github.com/RKrahl/pytest-dependency/pull/56 has more.

**Comment by chrisk314** [comment]
I just started using pytest-dependency, and it works great as is if you want to run the full test suite. However, I am often running individual tests within a suite to debug a specific issue. Without evaluating the upstream dependencies of the one test I actually want to run, my test just gets skipped. This is both surprising and clearly not helpful in this case.
